lang/c/build.sh

```diff
@@ -20,84 +20,82 @@
+function prepare_build()
+{
+  command_clean
+  execute mkdir -p "$build_dir"
+  execute pushd "$build_dir"
+  execute cmake "$root_dir" -DCMAKE_BUILD_TYPE=RelWithDebInfo
+  execute popd
 }
 
-function clean {
-  if [ -d $build_dir ]; then
-    find $build_dir | xargs chmod 755
-    rm -rf $build_dir
+function command_clean()
+{
+  if [ -d "$build_dir" ]; then
+    execute find "$build_dir" -exec chmod 755 {} +
+    execute rm -rf "$build_dir"
   fi
-  rm -f VERSION.txt
-  rm -f examples/quickstop.db
+  execute rm -f VERSION.txt
+  execute rm -f examples/quickstop.db
 }
 
-for target in "$@"
-do
-
-  case "$target" in
-
-    interop-data-generate)
-      prepare_build
-      make -C $build_dir
-      $build_dir/tests/generate_interop_data "../../share/test/schemas/interop.avsc"  "../../build/interop/data"
-      ;;
-
-    interop-data-test)
-      prepare_build
-      make -C $build_dir
-      $build_dir/tests/test_interop_data "../../build/interop/data"
-      ;;
-
-    lint)
-      echo 'This is a stub where someone can provide linting.'
-      ;;
-
-    test)
-      prepare_build
-      make -C $build_dir
-      make -C $build_dir test
-      ;;
+function command_interop-data-generate()
+{
+  prepare_build
+  execute make -C "$build_dir"
+  execute "$build_dir/tests/generate_interop_data" "$BUILD_ROOT/share/test/schemas/interop.avsc"  "$BUILD_ROOT/build/interop/data"
+}
 
-    dist)
-      prepare_build
-      cp ../../share/VERSION.txt $root_dir
-      make -C $build_dir docs
-      # This is a hack to force the built documentation to be included
-      # in the source package.
-      cp $build_dir/docs/*.html $root_dir/docs
-      make -C $build_dir package_source
-      rm $root_dir/docs/*.html
-      if [ ! -d $dist_dir ]; then
-        mkdir -p $dist_dir
-      fi
-      if [ ! -d $doc_dir ]; then
-        mkdir -p $doc_dir
-      fi
-      mv $build_dir/$tarball $dist_dir
-      cp $build_dir/docs/*.html $doc_dir
-      clean
-      ;;
+function command_interop-data-test()
+{
+    prepare_build
+    execute make -C "$build_dir"
+    execute "$build_dir/tests/test_interop_data" "$BUILD_ROOT/build/interop/data"
+}
 
-    clean)
-      clean
-      ;;
+function command_lint()
+{
+  echo 'This is a stub where someone can provide linting.'
+}
 
-    *)
-      echo "Usage: $0 {interop-data-generate|interop-data-test|lint|test|dist|clean}"
-      exit 1
-  esac
+function command_test()
+{
+  prepare_build
+  execute make -C "$build_dir"
+  execute make -C "$build_dir" test
+}
 
-done
+function command_dist()
+{
+  prepare_build
+  execute cp "$BUILD_ROOT/share/VERSION.txt" "$root_dir"
+  execute make -C "$build_dir" docs
+  # This is a hack to force the built documentation to be included
+  # in the source package.
+  execute cp "$build_dir"/docs/*.html "$root_dir/docs"
+  execute make -C "$build_dir" package_source
+  execute rm "$root_dir"/docs/*.html
+  if [ ! -d "$dist_dir" ]; then
+    execute mkdir -p "$dist_dir"
```

Comment:
If we're using `mkdir -p` we don't need to check if the dir already exists first.

Reply:
Roger. I followed the original build.sh scripts as much to the letter as possible. At this stage my objective was to be able to easily compare roiginal build shells cripts with the new ones.

Reply:
I understand but since you're adding `execute` doesn't that make it kind of moot?